### PR TITLE
Add tests for server's vm.register method + fix it to use new vm_service_lib with conditions

### PR DESCRIPTION
**Pull request by DanTup**
This adds tests for the vm.register method in the servers API and updates the *server* (but not devtools) to use the new vm_service_lib.

The builds are green now because the Dart SDK dev build doesn't have the changes yet, but the dev bots will go red once it updates until we land both this fix and also a fix to devtools (which I can't land until we've shipped the new server).

### CODE REVIEW

**Comment by DanTup** [comment]
@jacob314 @devoncarew these tests unexpectedly all went green (I expected the dev versions to fail) - minus a Windows process issue I'm logging at. I believe this is because we're using the Dev version of Dart here, and that does not yet have the VM change in it (whereas Flutter master does), so it will break on the next Dart dev roll. This PR does not yet include the fix, only the test.

However, it occurs to me that because in the repo we reference `devtools_server` from Pub, the tests are running against that version (and therefore if we break the server, the CI won't notice out until we publish it to pub and then update devtools to use it).

If we move the tests into devtools_server instead, we have a few different issues:

1. devtools_server currently cannot run in isolation (it calls `Isolate.resolvePackageUri(Uri(scheme: 'package', path: 'devtools/devtools.dart'))`
2. If in future we want to run real end-to-end tests through the server (to more accurately test what is running on the end users machine) we need to have a full devtools build (the current test could just a dummy build output folder since it only tests the server API)

There's also another complication.. In order to rev vm_service_lib, we need devtools + devtool_server *on pub* to match, so I think we would need to:

1. Update devtools_server to new vm_service_lib w/ fixes
2. Publish server
3. Update devtools_server version in devtools pubspec at the same time as bumping vm_server_lib and applying fixes there

Between 1 & 3, we would be unable to run locally using the `path` in pubspec.yaml for server, as the dependencies will conflict (they'll want different versions of vm_service_lib).

**Comment by DanTup** [comment]
I have changesets ready to raise for review (https://github.com/DanTup/devtools/commit/b5f4fea01e2366aa19be8c875a439b16f43c585d and https://github.com/DanTup/devtools/commit/bc183c36a3f51ec9d4fd3c1330e3dfcd12ccda89) if we want to do the steps above (1-4). I think it's the simplest solution (and not risky), though we'll need to do some testing after landing the last change before we publish a DevTools release (to ensure the `'ObjRef'` change is correct and we don't have any bugs against the new vm_service_lib.

File: packages/devtools_server/lib/src/server.dart [hunk 1]
```diff
@@ -254,7 +254,12 @@ Future<void> registerLaunchDevToolsService(
       }
     });
 
-    await service.registerService(launchDevToolsService, 'DevTools Server');
+    if (isVersionLessThan(await service.getVersion(), major: 3, minor: 22)) {
+      await service.callMethod('_registerService',
+          args: {'service': launchDevToolsService, 'alias': 'DevTools Server'});
+    } else {
+      await service.registerService(launchDevToolsService, 'DevTools Server');
```
Comment (devoncarew) [inline packages/devtools_server/lib/src/server.dart]: consider calling:

```
service.callMethod('registerService',
          args: {'service': launchDevToolsService, 'alias': 'DevTools Server'});
```

here.

File: packages/devtools_server/pubspec.yaml [hunk 2]
```diff
@@ -17,5 +17,5 @@ dependencies:
   pedantic: ^1.7.0
   shelf: ^0.7.4
   shelf_static: ^0.2.8
-  vm_service_lib: 3.21.0
+  vm_service_lib: 3.22.0
```
Comment (devoncarew) [inline packages/devtools_server/pubspec.yaml]: I believe that rolling this forward will cause NPEs in parts of the app (I did some spot checking of the memory page). We may want to de-couple rolling to this version with the support for the `_registerService` rename until we've had time to investigate any NPEs.

Comment (devoncarew) [reply packages/devtools_server/pubspec.yaml]: Ah, nevermind this - I'd through this change was in packages/devtools/pubspec.yaml.

**Comment by DanTup** [comment]
Alternative PR at https://github.com/flutter/devtools/pull/802 which  doesn't bump the vm_service_lib, but calls `_registerService` or `registerService` explicitly. This might be preferred to reduce the pressure to update the web app's dependency after landing this (since we know there are some bugs that will need fixing).

**Comment by DanTup** [comment]
Ditching this in favour of #802.

**Pull request closed without merging**